selfdrive/controls/lib/drive_helpers.py

```diff
@@ -52,3 +60,33 @@ def initialize_v_cruise(v_ego, buttonEvents, v_cruise_last):
       return v_cruise_last
 
   return int(round(clip(v_ego * CV.MS_TO_KPH, V_CRUISE_ENABLE_MIN, V_CRUISE_MAX)))
+
+
+def get_lag_adjusted_curvature(CP, v_ego, psis, curvatures, curvature_rates):
+  if len(psis) != CONTROL_N:
+    psis = [0.0 for i in range(CONTROL_N)]
+    curvatures = [0.0 for i in range(CONTROL_N)]
+    curvature_rates = [0.0 for i in range(CONTROL_N)]
+
+  # TODO this needs more thought, use .2s extra for now to estimate other delays
+  delay = CP.steerActuatorDelay + .2
+  current_curvature = curvatures[0]
+  psi = interp(delay, T_IDXS[:CONTROL_N], psis)
+  next_curvature_rate = curvature_rates[0]
+
+  # MPC can plan to turn the wheel and turn back before t_delay. This means
+  # in high delay cases some corrections never even get commanded. So just use
+  # psi to calculate a simple linearization of desired curvature
+  curvature_diff_from_psi = psi / (max(v_ego, 1e-1) * delay) - current_curvature
+  next_curvature = current_curvature + 2 * curvature_diff_from_psi
+
+  desired_curvature = next_curvature
+  desired_curvature_rate = next_curvature_rate
```

Comment:
seems these are just copies of next_curvature and next_curvature_rate. remove them or rename next_xxx to desired_xxx?